File: 409. Longest Palindrome.go
package main

func longestPalindrome(s string) int {

	freqMap := map[byte]int{}

	for i := 0; i < len(s); i++ {
		freqMap[s[i]]++
	}

	result := 0

	for _, value := range freqMap {
		result += value / 2 * 2
		if result%2 == 0 && value%2 == 1 {
			result += 1
		}
	}

	return result
}

File: 409. Longest Palindrome.py
class Solution:
    def longestPalindrome(self, s: str) -> int:
        
        fMap,result = defaultdict(int), 0
        for ch in s: fMap[ch] += 1

        curOdd = 0

        for k,v in fMap.items():
            if v % 2 == 0: 
                result += v
                continue
            if v > curOdd:
                if curOdd != 0:
                    result -= 1
                curOdd = v
                result += v
            else:
                result += v - 1

        return result

